woocommerce/woocommerce-ios
# HACK week: Replace AlamofireNetwork with a new network using URLSession

Part 1 of a POC for HACK week in December 2022.


### Description

This PR prepares the Networking layer to replace Alamofire with a native implementation of network handling with `URLSession`. Changes include:
- Replaced the use of `URLRequestConvertible` with the existing `Request`. 
- Copied Alamofire's `URLEncoding` & `HTTPMethod` since we want to reuse most of the logic here. Replace Alamofire's with the copied ones.
- Added an extension for `URLRequest` to construct a request with URL, HTTP method, and headers.
- Created a simple `SessionManager` that operates with a `URLSession`.
- Created a new network `URLSessionNetwork` and replaced `AlamofireNetwork` with it.

### Testing instructions

Log in to the app and do a smoke test on the features. Make sure that all network requests still work as before. Uploading media will be handled in a subsequent PR.

### Screenshots

N/A

---
